client/components/jetpack/connection-health/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { localizeUrl } from '@automattic/i18n-utils';
+import { useTranslate } from 'i18n-calypso';
+import { useState } from 'react';
+import Notice from 'calypso/components/notice';
+import NoticeAction from 'calypso/components/notice/notice-action';
+import { useDispatch } from 'calypso/state';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { useCheckJetpackConnectionHealth } from './use-check-jetpack-connection-health';
+
+interface Props {
+	siteId: number;
+}
+
+export const JetpackConnectionHealthBanner = ( { siteId }: Props ) => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const [ isErrorCheckJetpackConnectionHealth, setIsErrorCheckJetpackConnectionHealth ] =
+		useState( false );
+
+	const { data: jetpackConnectionHealth, isLoading: isLoadingJetpackConnectionHealth } =
+		useCheckJetpackConnectionHealth( siteId, {
+			onError: () => {
+				setIsErrorCheckJetpackConnectionHealth( true );
+			},
+		} );
+
+	const handleJetpackConnectionHealthLinkClick = () => {
+		dispatch(
+			recordTracksEvent( 'calypso_jetpack_connection_health_issue_click', {
+				type: 'default',
+			} )
+		);
+	};
+
+	if (
+		isLoadingJetpackConnectionHealth ||
+		isErrorCheckJetpackConnectionHealth ||
+		jetpackConnectionHealth?.is_healthy
+	) {
+		return;
+	}
+
+	dispatch(
+		recordTracksEvent( 'calypso_jetpack_connection_health_issue_view', {
+			type: 'default',
+		} )
+	);
```

Comment:
By doing that, when the user clicks on any tab, it will cause this to be triggered again, causing incorrect metrics.
I manage to fix it, by applying the following diff:
```
diff --git a/client/components/jetpack/connection-health/index.tsx b/client/components/jetpack/connection-health/index.tsx
index 0fcf37270b..8fda4bb9de 100644
--- a/client/components/jetpack/connection-health/index.tsx
+++ b/client/components/jetpack/connection-health/index.tsx
@@ -1,6 +1,6 @@
 import { localizeUrl } from '@automattic/i18n-utils';
 import { useTranslate } from 'i18n-calypso';
-import { useState } from 'react';
+import { useRef, useState } from 'react';
 import Notice from 'calypso/components/notice';
 import NoticeAction from 'calypso/components/notice/notice-action';
 import { useDispatch } from 'calypso/state';
@@ -13,6 +13,8 @@ interface Props {
 
 export const JetpackConnectionHealthBanner = ( { siteId }: Props ) => {
 	const translate = useTranslate();
+	const isFirstRender = useRef( true );
+
 	const dispatch = useDispatch();
 
 	const [ isErrorCheckJetpackConnectionHealth, setIsErrorCheckJetpackConnectionHealth ] =
@@ -41,11 +43,14 @@ export const JetpackConnectionHealthBanner = ( { siteId }: Props ) => {
 		return;
 	}
 
-	dispatch(
-		recordTracksEvent( 'calypso_jetpack_connection_health_issue_view', {
-			type: 'default',
-		} )
-	);
+	if ( isFirstRender.current ) {
+		isFirstRender.current = false;
+		dispatch(
+			recordTracksEvent( 'calypso_jetpack_connection_health_issue_view', {
+				type: 'default',
+			} )
+		);
+	}
 
 	return (
 		<Notice
```

Reply:
Thanks, good point. As we talked on Slack, I will change it to use the `TrackComponentView` component which is a mechanism created precisely for that purpose.